dotnet/src/SemanticKernel.Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Web;
+
+/// <summary>
+/// Skill to download web files.
+/// </summary>
+public class WebFileDownloadSkill : IDisposable
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Where to save file.
+        /// </summary>
+        public const string FilePath = "filePath";
+    }
+
+    private readonly ILogger _logger;
+    private readonly HttpClientHandler _httpClientHandler;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Constructor for WebFileDownloadSkill.
+    /// </summary>
+    /// <param name="logger">Optional logger.</param>
+    public WebFileDownloadSkill(ILogger<WebFileDownloadSkill>? logger = null)
+    {
+        this._logger = logger ?? NullLogger<WebFileDownloadSkill>.Instance;
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new HttpClient(this._httpClientHandler);
+    }
+
+    /// <summary>
+    /// Downloads a file to local storage.
```

Comment:
Nit/Suggestion: "downloads a file to a local file path". I know it's a little pedantic, sorry. I'd like to keep the terminology of 'storage' as contained as possible to the Memory.Storage namespace.

Reply:
Done